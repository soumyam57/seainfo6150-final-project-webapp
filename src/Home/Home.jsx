import React from 'react'
import styles from "./Home.module.css";
import GiftCategories from "../GiftCategories/GiftCategories.jsx"
import FeaturedGifts from "../FeaturedGifts/FeaturedGifts.jsx";
import Header from "../Header/Header.jsx";

const Home = () => {

    return (
        <div className={styles.homepage}>
            <GiftCategories/>
            <FeaturedGifts/> 
            <Header/>      
        </div>
    )
}

export default Home
