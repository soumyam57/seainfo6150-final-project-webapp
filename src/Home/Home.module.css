.homepage{ 
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
}

.title {
    box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5); 
    padding:2%;
    text-align: center;
    font-style: italic;
    font-weight: bold;
    text-transform: uppercase;
    color: gainsboro;
}



