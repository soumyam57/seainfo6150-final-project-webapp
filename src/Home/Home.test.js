import React from "react";
import { render } from "@testing-library/react";
import Home from "./Home.jsx";
import { BrowserRouter } from "react-router-dom";

it("renders the Home component correctly", () => {
  const { container } = render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
