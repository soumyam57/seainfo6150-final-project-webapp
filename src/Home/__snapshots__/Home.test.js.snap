// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Home component correctly 1`] = `
<div>
  <div
    class="homepage"
  >
    <div
      class="container"
    >
      <div
        class="flex"
      >
        <img
          alt=""
          class="image"
          src="wedding.jpg"
        />
        <p
          class="text"
        >
          <a
            class="text"
            href="/wedding"
          >
            Wedding & Party
          </a>
        </p>
      </div>
      <div
        class="flex"
      >
        <img
          alt=""
          class="image"
          src="christmas.jpg"
        />
        <p
          class="text"
        >
          <a
            class="text"
            href="/festival"
          >
            Holidays & Festival
          </a>
        </p>
      </div>
      <div
        class="flex"
      >
        <img
          alt=""
          class="image"
          src="dolls.jpg"
        />
        <p
          class="text"
        >
          <a
            class="text"
            href="/kids"
          >
            Kids
          </a>
        </p>
      </div>
      <div
        class="flex"
      >
        <img
          alt=""
          class="image"
          src="Birthday.jpg"
        />
        <p
          class="text"
        >
          <a
            class="text"
            href="/birthday"
          >
            Birthday
          </a>
        </p>
      </div>
      <div
        class="flex"
      >
        <img
          alt=""
          class="image"
          src="getwell.jpg"
        />
        <p
          class="text"
        >
          <a
            class="text"
            href="/getwell"
          >
            Get Well Soon
          </a>
        </p>
      </div>
    </div>
    <div>
      <h1
        class="title"
      >
        Popular Today
      </h1>
      <div
        class="container"
      >
        <div
          class="grid"
        >
          <div
            class="div"
          >
            <img
              alt=""
              class="image"
              src="livetree.jpg"
            />
            <a
              class="text"
              href="/festival/livetree"
            >
              Christmas Reindeer Live Tree
            </a>
          </div>
          <div
            class="div"
          >
            <img
              alt=""
              class="image"
              src="photoframe.jpg"
            />
            <a
              class="text"
              href="/wedding/photoholder"
            >
              Couple Gifts Photo Holder
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
